### ISSUE TITLE
can't import tensorflow.keras properly

### ISSUE THREAD

**Issue by DequanZhu** [tensorflow/tensorflow#26813]
I,m writing my code in vscode edit with tensorflow=1.13.1 version and anaconda virtual environment. But when I write 'from tensorflow.keras import layers',it give me a warning:
"unresolved import 'tensorflow.keras'(unresolved import)". 
The code can run as I expected,no errors. But because tensorflow.keras can't be imported properly,the auto-completion and intelligent hint function can't work,I need to search the function's usage everytime. I have thought it's the problem of vscode, but the problem came as well when I use pycharm IDE. Have anyone has the same problem? Is there anyone can help me?

**Comment by mr-ubik** [comment]
The problem is present also in 2.0. Apparently, Jedi (or Pyls) cannot resolve Keras import.

**Comment by AlexMetsai** [comment]
Works fine for me, with both ways of importing
"from tensorflow.python.keras import layers"
and
"from tensorflow.keras import layers"

Can you give us a little more info about your configuration?

**Comment by mr-ubik** [comment]
In my case I am using the 2.0 Nightly, the issue is related to autocomplete either using Jedi or Pyls (Python Language Server) with either Vim/Neovim/Pycharm or VS Code.
The issue is present also if linting with pylint as it cannot see Keras marking it as an erroneous import. The problem is common to all 4 machines we use in our lab. The only peculiar thing is that we run Arch Linux but Tensorflow is installed in a separate virtualenv. Every other module seems to work fine, Keras is the only problematic one.

**Comment by DequanZhu** [comment]
> Works fine for me, with both ways of importing
> "from tensorflow.python.keras import layers"
> and
> "from tensorflow.keras import layers"
> 
> Can you give us a little more info about your configuration?

I works on windows10 platform, and I use the python plugin for vscode. It seems that just tensorflow's keras module can't be imported properly, i.e. vscode's automatic completion function didn't work but other modules works fine.

**Comment by AlexMetsai** [comment]
If the issue is related to autocomplete related to your IDEs, then maybe this isn't a tensorflow issue.
Can you import from a python interpreter that does NOT use any autocomplete method? If you already tried that, does the problem persist?

**Comment by mr-ubik** [comment]
Importing keras in an interpreter works but all linters and autocomplete providers cannot resolve it. Unless it's something shared by all of them I would guess it's TF

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by iLeW** [comment]
The automatic completion never worked for me either. I'm using PyCharm (professional edition).

**Comment by darophi** [comment]
You can use the v1 api like so:
```python

from tensorflow._api.v1.keras import Sequential
from tensorflow._api.v1.keras.layers import Dense
```
or the tensorflow.python api like so:

```python
from tensorflow.python.keras import Sequential
from tensorflow.python.keras.layers import Dense
```

**Comment by ymodak** [comment]
Closing this issue since it is IDE related. Feel free to reopen if have any questions. Thanks!

**Comment by DequanZhu** [comment]
It solved my problem, thank you!

At 2019-04-17 18:00:06, "Raphael Neumann" <notifications@github.com> wrote:


You can use the v1 api like so:

from tensorflow._api.v1.keras import Sequential
from tensorflow._api.v1.keras.layers import Dense

or the tensorflow.python api like so:

from tensorflow.python.keras import Sequential
from tensorflow.python.keras.layers import Dense

**Comment by kong13661** [comment]
It seems because auto-complete won't scan the whole code of a library. I open `__init__.py` file whose path is `/usr/local/lib/python3.6/dist-packages/tensorflow/__init__.py` in my computer, and add one line `from tensorflow.python import keras`. It works for me. And you can find the path of the file by using finding definition of your IDE. I think it's a temporary way.

**Comment by astier** [comment]
@kong13661 so it is a tf issue.

**Comment by yurnery** [comment]
this is still a problem for me, it cann't auto-complete the code.

**Comment by ajmssc** [comment]
this is reproducible outside of IDEs using pylint
```
python -m pylint --disable=all --enable=E0401 model.py
```

**Comment by astier** [comment]
Please reopen. This is a real issue.

**Comment by dantengdekaoya** [comment]
Please reopen. This is a real issue.

**Comment by janosh** [comment]
I third @astier. Getting `unresolved import 'tensorflow.keras.layers'` with

```py
from tensorflow.keras.layers import Conv2D, Dense, Flatten, MaxPool2D
```

Importing from `tensorflow.python.keras.layers` makes no difference. This is in TF r2.1.

**Comment by pjsjongsung** [comment]
Using tensorflow 2.1.0 in VScode.
I'm having the same issue.

**Comment by Adilkhan510** [comment]
Having the same issue. Please reopen this

**Comment by ximalayashanjiao** [comment]
Having the same issue. Please reopen this

**Comment by marewi** [comment]
same issue with VScode 1.42.1 & tensorflow 2.0.1

**Comment by soheilazangeneh** [comment]
Same issue TensorFlow 2.0 in PyCharm.

**Comment by dantengdekaoya** [comment]
If you upgrade pycharm to 2019.3 version, this issue may can be solved.

Soheila Zangeneh <notifications@github.com>于2020年2月20日 周四上午10:25写道：

**Comment by adriangb** [comment]
Same issue here. All other packages work fine.

**Comment by soheilazangeneh** [comment]
> If you upgrade pycharm to 2019.3 version, this issue may can be solved. Soheila Zangeneh <notifications@github.com>于2020年2月20日 周四上午10:25写道：
> […](#)

Worked for me! Thanks!

**Comment by ShreyasRP** [comment]
Try removing the prefix "tensorflow".
Eg: from keras import layers

**Comment by dskkato** [comment]
When I switched to use flake8 instead of pylint, all error message related to importing keras has gone. If you try, open VScode setting page and serach `Linting Enabled`, and

1. Uncheck `Python › Linting: Pylint Enabled`
2. Check `Python › Linting: Flake8 Enabled`

**Comment by raspstephan** [comment]
I opened an issue for this on the vscode-python page: https://github.com/microsoft/vscode-python/issues/10598

**Comment by sunn1melon** [comment]
> 
> 
> When I switched to use flake8 instead of pylint, all error message related to importing keras has gone. If you try, open VScode setting page and serach `Linting Enabled`, and
> 
>     1. Uncheck `Python › Linting: Pylint Enabled`
> 
>     2. Check `Python › Linting: Flake8 Enabled`
> 
> 
> Other solution?
> 
> [#15736 (comment)](https://github.com/tensorflow/tensorflow/issues/15736#issuecomment-354642939)

That finally worked for me! Thanks a lot for that hint!

**Comment by mayankgupta1905** [comment]
Working with the 2.1.0. The problem persists only with the TensorFlow modules.

**Comment by Mrmachiner** [comment]
uninstall extension Visual Studio IntelliCode works fine for me

**Comment by mihaimaruseac** [comment]
Locking conversation to drive people encountering new issues with newer code to open new issue reports (filling in template) and to help people encountering the same issue find the solution here
